index.js

```diff
@@ -36,89 +39,89 @@ instance.prototype.init = function() {
 	if (yt_dir_path) {
 		fs.readFile(secrets_file_path, (err, config_file) => {
 			if (err) {
-				self.log("warn", "Cannot load app OAuth credentials, " + err);
-				self.status(self.STATUS_ERROR, "Cannot load app OAuth credentials, " + err);
+				self.log('warn', 'Cannot load app OAuth credentials: ' + err);
+				self.status(self.STATUS_ERROR, 'Cannot load app OAuth credentials: ' + err);
+				return;
+			}
+
+			try {
+				var config_json   = JSON.parse(config_file);
+				var client_id     = config_json["web"]["client_id"];
+				var client_secret = config_json["web"]["client_secret"];
+				var redirect_url  = config_json["web"]["redirect_uris"][0];
+
+				self.yt_api_handler = new Youtube_api_handler(client_id, client_secret, redirect_url, scopes, token_path);
+
+			} catch (err) {
+				self.log('warn', 'Cannot parse app OAuth credentials: ' + err);
+				self.status(self.STATUS_ERROR, 'Cannot parse app OAuth credentials: ' + err);
 				return;
 			}
 
-			var config_json   = JSON.parse(config_file);
-			var client_id     = config_json["web"]["client_id"];
-			var client_secret = config_json["web"]["client_secret"];
-			console.log("client_seceret: " + client_secret);
-			var redirect_url  = config_json["web"]["redirect_uris"][0];
-
-			self.yt_api_handler = new Youtube_api_handler(client_id, client_secret, redirect_url, scopes, token_path);
-			if (fs.existsSync(token_path)) {
-				if (fs.lstatSync(token_path).isFile()) {
-					console.log("Token file (token.json) is provided");
-
-					fs.readFile(token_path, (error, token) => {
-						if (error) {
-							let promise = self.yt_api_handler.oauth_login()
-							promise.then(acces_credentials => {
-								self.yt_api_handler.oauth2client.credentials = acces_credentials;
-								self.yt_api_handler.create_yt_service();
-								let get_broadcasts = self.yt_api_handler.get_all_broadcasts();
-								get_broadcasts.then(streams_dict => {
-									self.yt_api_handler.streams_dict = streams_dict;
-									console.log(self.yt_api_handler.streams_dict);
-									self.actions();
-								});
-								get_broadcasts.catch(console.error);
-							});
-							promise.catch(console.error);
-						} else {
-							self.yt_api_handler.oauth2client.credentials = JSON.parse(token);
-							self.yt_api_handler.create_yt_service();
-							let get_broadcasts = self.yt_api_handler.get_all_broadcasts();
-							get_broadcasts.then(streams_dict => {
-								self.yt_api_handler.streams_dict = streams_dict;
-								console.log(self.yt_api_handler.streams_dict);
-								self.actions();
-							});
-							get_broadcasts.catch(console.error);
-						}
-					});
+			fs.readFile(token_path, (err, token) => {
+				var credentials;
+
+				if (!err) {
+					try {
+						credentials = JSON.parse(token);
+					} catch (err2) {
+						err = err2;
+					}
+				}
+
+				if (!err) {
```

Comment:
These two if (!err) { could have been combined by removing `}` and `if (!err) {`

Reply:
I'll merge it anyways

Reply:
This is intentional - the err variable gets set in the catch branch above.